src/test/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/DataSourceUtilsUnitTest.java

```diff
@@ -30,40 +34,223 @@
 
         final ArrayList<Object[]> testArgs = new ArrayList<>();
 
-        final ArrayList<Integer> baseArgs = new ArrayList<>();
-        baseArgs.add(DataSourceUtils.MIN_MAJOR_VERSION_NUMBER);
-        baseArgs.add(DataSourceUtils.MIN_MINOR_VERSION_NUMBER);
-        baseArgs.add(DataSourceUtils.MIN_YEAR_RELEASED);
-        baseArgs.add(DataSourceUtils.MIN_MONTH_RELEASED);
-        baseArgs.add(DataSourceUtils.MIN_DAY_RELEASED);
+        // =============================================================
+        // First do tests that check the major / minor version numbers:
+        // -------------------------------------------------------------
 
-        for ( int offset = -1 ; offset < 2; ++offset ) {
-            for ( int i = 0; i < baseArgs.size(); ++i ) {
+        final int MINOR_IT_MAX_VAL = 20;
 
-                final ArrayList<Object> argList = new ArrayList<>();
+        // major < MIN_MAJOR (minor shouldn't matter)
+        for (int minor = 0; minor < MINOR_IT_MAX_VAL; ++minor) {
+            testArgs.add(
+                    new Object[]{
+                            DataSourceUtils.MIN_MAJOR_VERSION_NUMBER - 1, minor, DataSourceUtils.MIN_DATE, false
+                    }
+            );
+        }
+
+        // major == MIN_MAJOR (minor does matter)
+        for (int minor = 0; minor < MINOR_IT_MAX_VAL; ++minor) {
+            testArgs.add(
+                    new Object[]{
+                            DataSourceUtils.MIN_MAJOR_VERSION_NUMBER, minor, DataSourceUtils.MIN_DATE,
+                            (minor == DataSourceUtils.MIN_MINOR_VERSION_NUMBER || minor == DataSourceUtils.MAX_MINOR_VERSION_NUMBER)
+                    }
+            );
+        }
 
-                argList.addAll(baseArgs.subList(0, i));
-                argList.add(baseArgs.get(i) + offset);
+        // major > MIN_MAJOR and < MAX_MAJOR (minor shouldn't matter)
+        // Can't currently test here - version numbers use integer values and right now
+        // MIN_MAJOR_VERSION_NUMBER == MAX_MAJOR_VERSION_NUMBER
+        // TODO: Add in test when possible.
+
+        // major == MAX_MAJOR (minor does matter)
+        for (int minor = 0; minor < MINOR_IT_MAX_VAL; ++minor) {
+            testArgs.add(
+                    new Object[]{
+                            DataSourceUtils.MAX_MAJOR_VERSION_NUMBER, minor, DataSourceUtils.MAX_DATE,
+                            (minor == DataSourceUtils.MIN_MINOR_VERSION_NUMBER || minor == DataSourceUtils.MAX_MINOR_VERSION_NUMBER)
```

Comment:
I don't think this condition is right either. We know that major == MAX_MAJOR here, so minor needs to be <= MAX_MINOR. If major also happens to == MIN_MAJOR (which it does when MIN_MAJOR == MAX_MAJOR), then, then minor must also be >= MIN_MINOR. So here I think we want:

`minor <= DataSourceUtils.MAX_MINOR_VERSION_NUMBER && (major != DataSourceUtils.MIN_MAJOR_VERSION_NUMBER || minor >= DataSourceUtils.MIN_MINOR_VERSION_NUMBER)`

Reply:
Yeah - this was tailored to the current values of the MIN_ and MAX_ minor versions.

These tests have been refactored as per some of your other comments, but I'll retain the new conditions.